lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,373 @@
+            surface_pressure.convert_units('hPa')
+
+            return self.model_level_temperature_lapse_rate(
+                cube, sites, neighbours, pressure_on_height_levels,
+                surface_pressure, temperature_on_height_levels)
+
+        raise AttributeError('Unknown method "{}" passed to {}.'.format(
+            self.method, self.__class__.__name__))
+
+    @staticmethod
+    def _build_coordinates(latitudes, longitudes, site_ids, gmtoffsets):
+        """
+        Construct coordinates for the irregular iris.Cube containing site data.
+        A single dimensional coordinate is created using the running order,
+        whilst the non-monotonically increasing coordinates (e.g. bestdata_id)
+        are stored in AuxilliaryCoordinates.
+
+        Args:
+        -----
+        latitudes : list
+            A list of latitudes ordered to match the sites OrderedDict.
+
+        longitudes : list
+            A list of longitudes ordered to match the sites OrderedDict.
+
+        site_ids   : list
+            A list of bestdata site_ids ordered to match the sites OrderedDict.
+
+        gmtoffsets : list
+            A list of gmt off sets in hours ordered to match the sites
+            OrderedDict.
+
+        Returns:
+        --------
+        Creates iris.DimCoord and iris.AuxCoord objects from the provided data
+        for use in constructing new cubes.
+
+        """
+        indices = DimCoord(np.arange(len(latitudes)), long_name='index',
+                           units='1')
+        bd_ids = AuxCoord(site_ids, long_name='bestdata_id', units='1')
+        latitude = AuxCoord(latitudes, standard_name='latitude',
+                            units='degrees')
+        longitude = AuxCoord(longitudes, standard_name='longitude',
+                             units='degrees')
+        gmtoffset = AuxCoord(gmtoffsets, long_name='gmtoffset',
+                             units='hours')
+        return indices, bd_ids, latitude, longitude, gmtoffset
+
+    def make_cube(self, cube, data, sites):
+        """
+        Construct and return a cube containing the data extracted from the
+        grids by the desired method for the sites provided.
+
+        Args:
+        -----
+        cube : iris.cube.Cube
+            The original diagnostic cube from which data has been extracted.
+
+        data : numpy.array
+            Array of diagnostic values extracted for the defined sites.
+
+        sites : dict
+            A dictionary containing the properties of spotdata sites.
+
+        Returns:
+        --------
+        cube : iris.cube.Cube
+            An irregularly (i.e. non-gridded) cube of diagnostic data extracted
+            at the spotdata sites.
+
+        """
+        latitudes = [site['latitude'] for site in sites.itervalues()]
+        longitudes = [site['longitude'] for site in sites.itervalues()]
+        gmtoffsets = [site['gmtoffset'] for site in sites.itervalues()]
```

Comment:
I know we have just talked about pedantic reviews, but `utcoffset` is strictly speaking more accurate (`GMT` no longer really exists, and when it does it can be used in two different ways).